CMakeLists.txt

```diff
@@ -0,0 +1,153 @@
+cmake_minimum_required(VERSION 3.23)
+project(
+    cppfront
+    LANGUAGES CXX
+    VERSION 0.1.0
+)
+
+##
+# Project options
+
+option(
+    CPPFRONT_INSTALL_RULES "Enable install rules for cppfront" "${PROJECT_IS_TOP_LEVEL}"
+)
+
+##
+# Target definition for cppfront executable
+
+add_executable(cppfront_cppfront source/cppfront.cpp)
+add_executable(cppfront::cppfront ALIAS cppfront_cppfront)
+set_target_properties(
+    cppfront_cppfront
+    PROPERTIES
+    OUTPUT_NAME cppfront
+    EXPORT_NAME cppfront
+)
+
+target_compile_features(cppfront_cppfront PRIVATE cxx_std_20)
+target_sources(
+    cppfront_cppfront
+    PRIVATE
+    FILE_SET HEADERS
```

Comment:
Speaking of prefixes, are file sets global or directory scoped?

Reply:
Target scoped:
> Targets have zero or more named file sets.
> [...]
> The following target properties are set by target_sources(FILE_SET), but they should not generally be manipulated directly:
> -- https://cmake.org/cmake/help/latest/command/target_sources.html#file-sets